src/daft-local-execution/src/run.rs

```diff
@@ -46,6 +47,13 @@ impl LocalPartitionIterator {
 #[cfg_attr(feature = "python", pyclass(module = "daft.daft"))]
 pub struct NativeExecutor {
     local_physical_plan: Arc<LocalPhysicalPlan>,
+    cancel: CancellationToken,
```

Comment:
The tasks spawned on the execution runtime handle currently only abort if they try a `send` and see that the channel is closed (this is something we did in the native runner pr). However, they should also just abort if the native executor gets dropped.